File: web/src/api/prelabel.ts
import { api, type ApiEnvelope } from "./client";

export type PreLabeledRegion = {
  designator: string;
  bbox: [number, number, number, number];
  confidence: number;
  side: "top" | "bottom";
};

export type PreLabelRun = {
  id: string;
  project_id: string;
  side: "top" | "bottom";
  regions: PreLabeledRegion[];
  created_at: string;
};

export async function runPreLabel(
  projectId: string,
  side: "top" | "bottom",
): Promise<PreLabelRun> {
  const r = await api.post<ApiEnvelope<PreLabelRun>>(
    `/api/projects/${projectId}/llm/prelabel`,
    null,
    { params: { side } },
  );
  return r.data.data;
}

export async function getPreLabel(
  projectId: string,
  side: "top" | "bottom",
): Promise<PreLabelRun | null> {
  try {
    const r = await api.get<ApiEnvelope<PreLabelRun>>(
      `/api/projects/${projectId}/llm/prelabel`,
      { params: { side } },
    );
    return r.data.data;
  } catch (e: unknown) {
    // 404 → no pre-label yet; surface as null
    const status = (e as { response?: { status?: number } }).response?.status;
    if (status === 404) return null;
    throw e;
  }
}

File: src/indusia_visual_editor/schemas/prelabel.py
"""Request/response schemas for the pre-label route."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

from indusia_visual_editor.services.llm.schemas import PreLabeledRegion


class PreLabelRunRead(BaseModel):
    """Response shape for POST/GET `/api/projects/{id}/llm/prelabel`."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    side: Literal["top", "bottom"]
    regions: list[PreLabeledRegion]
    created_at: datetime
